include/gsl/span

```diff
@@ -330,8 +345,11 @@ public:
 
     using size_type = index_type;
 
-    constexpr static const index_type extent = Extent;
-
+#if defined(GSL_USE_STATIC_CONSTEXPR_WORKAROUND)
```

Comment:
This is an odd choice. Why not simply provide the out-of-class definition for C++ < 17?

Reply:
From memory: one of the compilers gave me a complaint about it. Plus, this seemed like the very well-known-but-ugly moral equivalent of the C++17 constexpr (i.e. it has a very good chance of actually getting optimized away). It's only a temporary band-aid, so I'm not overly attached, but nor did I want to spend too much time perfecting it.

Reply:
Fair enough. I'd suggest fixing the underlying type of the enum so `extent` always has the same type:
```c++
enum : index_type { extent = Extent };
```

Reply:
Yep, sounds good!